impeller/golden_tests/golden_playground_test_mac.cc

```diff
@@ -0,0 +1,139 @@
+  // Make sure there are no slashes in the test name.
+  std::replace(result.begin(), result.end(), '/', '_');
+  return result;
+}
+
+std::string GetGoldenFilename() {
+  return GetTestName() + ".png";
+}
+
+bool SaveScreenshot(std::unique_ptr<testing::MetalScreenshot> screenshot) {
+  if (!screenshot || !screenshot->GetBytes()) {
+    return false;
+  }
+  std::string test_name = GetTestName();
+  std::string filename = GetGoldenFilename();
+  testing::GoldenDigest::Instance()->AddImage(
+      test_name, filename, screenshot->GetWidth(), screenshot->GetHeight());
+  return screenshot->WriteToPNG(
+      testing::WorkingDirectory::Instance()->GetFilenamePath(filename));
+}
+}  // namespace
+
+struct GoldenPlaygroundTest::GoldenPlaygroundTestImpl {
+  GoldenPlaygroundTestImpl()
+      : screenshoter_(new testing::MetalScreenshoter()) {}
+  std::unique_ptr<testing::MetalScreenshoter> screenshoter_;
+  ISize window_size_ = ISize{1024, 768};
+};
+
+GoldenPlaygroundTest::GoldenPlaygroundTest()
+    : pimpl_(new GoldenPlaygroundTest::GoldenPlaygroundTestImpl()) {}
+
+void GoldenPlaygroundTest::SetUp() {
+  if (GetBackend() != PlaygroundBackend::kMetal) {
+    GTEST_SKIP_("GoldenPlaygroundTest doesn't support this backend type.");
+    return;
+  }
+
+  std::string test_name = GetTestName();
+  if (test_name == "impeller_Play_AiksTest_CanRenderTiledTexture_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_CanRenderLinearGradientWithOverlappingStops_"
+          "Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_CanRenderLinearGradientManyColors_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_CanRenderLinearGradientWayManyColors_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_CanRenderLinearGradientManyColorsUnevenStops_"
+          "Metal" ||
+      test_name == "impeller_Play_AiksTest_CanRenderRadialGradient_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_CanRenderRadialGradientManyColors_Metal" ||
+      test_name == "impeller_Play_AiksTest_CanRenderSweepGradient_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_CanRenderSweepGradientManyColors_Metal" ||
+      test_name == "impeller_Play_AiksTest_CanPictureConvertToImage_Metal" ||
+      test_name == "impeller_Play_AiksTest_TextFrameSubpixelAlignment_Metal" ||
+      test_name == "impeller_Play_AiksTest_ColorWheel_Metal" ||
+      test_name == "impeller_Play_AiksTest_SolidStrokesRenderCorrectly_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_GradientStrokesRenderCorrectly_Metal" ||
+      test_name ==
+          "impeller_Play_AiksTest_"
+          "CoverageOriginShouldBeAccountedForInSubpasses_Metal" ||
+      test_name == "impeller_Play_AiksTest_SceneColorSource_Metal" ||
+      test_name == "impeller_Play_AiksTest_CanRenderLinearGradient_Metal") {
+    GTEST_SKIP_(
+        "GoldenPlaygroundTest doesn't support interactive playground tests "
+        "yet.");
+  }
```

Comment:
It'd be nice if we could somehow check if the test actually used `ImGui` or not. This will be brittle, but it's probably just temporary right?

Reply:
I looked into it, the problem is that we have to call `GTEST_SKIP` inside of `SetUp` so there's no way (at runtime) to know if a test is going to use ImGui in `SetUp` =T

I think we should probably abolish any ImGui tests since they can't be automated.

Reply:
Alternatively we move them into some other target that makes it clear they'r enot automated tests.

Reply:
Yea I was thinking about something similar, maybe making the target something like `impeller_playground`.  I'm not sure if we want to just move everything over to goldens though.  In which case there would be no need for that.  I'm still trying to feel out the problem.